### PR TITLE
Support more icons for media players

### PR DESCRIPTION
**Pull request by felipecrs**
## Breaking change

N/A

## Proposed change

This PR is a follow-up of:

- https://github.com/home-assistant/core/pull/73832

There, I realized that many integrations explicitly set `icon` of entities to `speaker` because the frontend does not render a speaker by default for media players of device class `speaker`.

This improves it a little, by adding some more icons to them.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

N/A

## Additional information

- This PR fixes or closes issue: N/A
- This PR is related to issue or discussion:
  - https://github.com/home-assistant/core/pull/73832
  - https://github.com/Templarian/MaterialDesign/issues/6668
- Link to documentation pull request: N/A

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/common/entity/domain_icon.ts [hunk 1]
```diff
@@ -125,7 +134,30 @@ export const domainIconWithoutDefault = (
       }
 
     case "media_player":
-      return compareState === "playing" ? mdiCastConnected : mdiCast;
+      switch (stateObj?.attributes.device_class) {
+        case "receiver":
+          return compareState === "off" ? mdiAudioVideoOff : mdiAudioVideo;
+        case "speaker":
+          return compareState === "off" ? mdiSpeakerOff : mdiSpeaker;
```
[the remark this answers is not in the document]
Comment (bramkragten) [reply src/common/entity/domain_icon.ts:141]: The last one looks like a playing speaker, but it appears to be for a wireless speaker... I would say it fits a playing speaker pretty well

[the remark this answers is not in the document]
Comment (bramkragten) [reply src/common/entity/domain_icon.ts:141]: I would keep the logic we have today, and change the icon based on if the media player is playing or not, additionally we can add other icons for different states, but atleast have those 2.

[the remark this answers is not in the document]
Comment (felipecrs) [reply src/common/entity/domain_icon.ts:141]: @bramkragten new icons were added to MDI for speaker playing and paused:

- https://github.com/Templarian/MaterialDesign/issues/6668

So, how about we start with the Off variant only and as soon as the new MDI gets released, we add more variants to speaker?

[the remark this answers is not in the document]
Comment (bramkragten) [reply src/common/entity/domain_icon.ts:141]: Nice! I don't think we should change it now though, as we now change the icon based on playing or not, and we would change that to on/off now... I think that will cause confusion.

**Comment by felipecrs** [comment]
Waiting for the newer version of MDI to be released which will contain:

- https://github.com/Templarian/MaterialDesign/issues/6668

**Comment by felipecrs** [comment]
@frenck this is ready to review. 👍

**Comment by felipecrs** [comment]
Once this gets merged, lots of integrations that provide media_player can be refactored to use only `device_class`, instead of referencing custom (and therefore static -- not state based) icons.

**Review by bramkragten — DISMISSED** [review]

**Comment by bramkragten** [comment]
The new icons are not in the 6.9.96 release yet:
<https://dev.materialdesignicons.com/changelog#version-6.9.96>

**Comment by felipecrs** [comment]
Oh wow! I'm sorry! I got bamboozled with their website:

https://user-images.githubusercontent.com/29582865/177312418-4350b00c-82dc-4ba6-82b2-a4800ff43ce1.mp4

**Comment by bramkragten** [comment]
Looks like it will be part of the next major release, 7.x.x

**Review by bramkragten — APPROVED** [review]

**Pull request merged**